crates/wdl-engine/src/value.rs

```diff
@@ -1998,6 +2061,94 @@ impl fmt::Display for Struct {
     }
 }
 
+/// An enum variant value.
+///
+/// A variant enum is the name of the enum variant and the type of the enum from
+/// which that variant can be looked up.
+///
+/// This type is cheaply clonable.
+#[derive(Debug, Clone)]
+pub struct EnumVariant {
+    /// The type of the enum containing this variant.
+    enum_ty: EnumType,
+    /// The name of the variant.
+    name: Arc<String>,
```

Comment:
I still think we could probably store the index of the variant rather than the name here, since we have the associated `EnumType`, which stores the names.